pydecorate/decorator_base.py

```diff
@@ -748,21 +755,26 @@ def _insert_RGBA_image(self, img, box):
 
 
 def _create_colorbar_image(
-    colormap, minval, maxval, scale_height, scale_width, is_vertical
+    colormap, min_cmap_value, max_cmap_value, scale_height, scale_width, is_vertical
 ):
     if TImage is None:
         raise ImportError(
             "Missing 'trollimage' dependency. Required colorbar creation."
         )
 
     if is_vertical:
+        # Image arrays have row 0 being the top of the image so we must flip all calculations
         linedata = np.ones((scale_width, 1)) * np.arange(
-            minval, maxval, float(maxval - minval) / scale_height
+            max_cmap_value,
+            min_cmap_value,
+            float(min_cmap_value - max_cmap_value) / scale_height,
```

Comment:
`np.linspace(max_cmap_value, min_cmap_value, scale_height, endpoint=False)` is equivalent to the `arange` and also clearer imo

Reply:
I have a feeling when these were originally written the endpoint keyword argument didn't exist...or the person didn't know about linspace. Let me do some tests to make sure it produces the same then but yeah agreed it should overall be clearer.